internal/kafka/internal/services/data_plane_cluster_test.go

```diff
@@ -77,6 +77,79 @@ func Test_DataPlaneCluster_UpdateDataPlaneClusterStatus(t *testing.T) {
 				return NewDataPlaneClusterService(sampleValidApplicationConfigForDataPlaneClusterTest(clusterService))
 			},
 		},
+		{
+			name:          "An error is returned when svcErr != nil",
+			clusterID:     testClusterID,
+			clusterStatus: nil,
+			dataPlaneClusterServiceFactory: func() *dataPlaneClusterService {
+				clusterService := &ClusterServiceMock{
+					FindClusterByIDFunc: func(clusterID string) (*api.Cluster, *errors.ServiceError) {
+						return nil, &errors.ServiceError{}
+					},
+				}
+				return NewDataPlaneClusterService(sampleValidApplicationConfigForDataPlaneClusterTest(clusterService))
+			},
+			wantErr: true,
+		},
+		{
+			name:          "An error is returned when Status: api.ClusterFailed ",
+			clusterID:     testClusterID,
+			clusterStatus: nil,
+			dataPlaneClusterServiceFactory: func() *dataPlaneClusterService {
+				clusterService := &ClusterServiceMock{
+					FindClusterByIDFunc: func(clusterID string) (*api.Cluster, *errors.ServiceError) {
+						return &api.Cluster{
+							Meta: api.Meta{
+								ID: "id",
+							},
+							ClusterID: clusterID,
+							Status:    api.ClusterFailed,
+						}, nil
+					},
+				}
+				return NewDataPlaneClusterService(sampleValidApplicationConfigForDataPlaneClusterTest(clusterService))
+			},
+			wantErr: false,
+		},
+		{
+			name:      "an error is returned when fleetShardOperatorReady is false",
```

Comment:
This description is too implementation specific. An alternative name might be:
An error is returned when the fleet shard operator is not ready